crates/oxc_linter/src/rules/eslint/no_label_var.rs

```diff
@@ -0,0 +1,69 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_label_var_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("eslint(no-label-var): Found identifier with same name as label.")
```

Comment:
```suggestion
    OxcDiagnostic::warn("eslint(no-label-var): Found an identifier with the same name as a label.")
```

Reply:
Adjusted and then re-adjusted as I added a the label spans with text. Maybe I went a bit overboard but it looks nice:

```
  × eslint(no-label-var): Found identifier 'x' with the same name as a label.
   ╭─[foo.js:3:5]
 2 │
 3 │ var x = foo;
   ·     ┬
   ·     ╰── Identifier 'x' found here.
 4 │
 5 │
 6 │ function bar() {
 7 │ x:
   · ┬
   · ╰── Label with the same name.
 8 │   for (;;) {
   ╰────
```